wayland-client/src/lib.rs

```diff
@@ -72,6 +82,22 @@ pub trait Proxy: Sized {
     /// used by code generated by wayland-scanner.
     fn from_id(conn: &Connection, id: ObjectId) -> Result<Self, InvalidId>;
 
+    /// Send a request for this object.
+    ///
+    /// It is an error to use this function on requests that create objects; use
+    /// [Proxy::send_request_create] for such requests.
+    fn send_request_nocreate(&self, req: Self::Request) -> Result<(), InvalidId>;
+
+    /// Send a request for this object that creates another object.
+    ///
+    /// It is an error to use this function on requests that do not create objects; use
+    /// [Proxy::send_request_nocreate] for such requests.
+    fn send_request_create<I: Proxy>(
+        &self,
+        req: Self::Request,
+        data: Arc<dyn ObjectData>,
+    ) -> Result<I, InvalidId>;
+
```

Comment:
If we want to stick with the conventions from libwayland, we could name them `send_request` and `send_constructor`.